glue/core/data_derived.py

```diff
@@ -194,3 +194,181 @@ def compute_histogram(self, *args, **kwargs):
+                             .format(original_data.ndim))
+
+        if hasattr(original_data, 'coords'):
+            if original_data.coords is None:
+                self._coords = None
+            else:
+                slices_0 = [slice(None) if idx == 0 else idx for idx in slices]
+                self._coords = SlicedLowLevelWCS(original_data.coords, slices_0)
+
+        self._original_data = original_data
+        self._cid_to_original_cid = {}
+        self._original_cid_to_cid = {}
+        self.slices = slices
+
+    @property
+    def slices(self):
+        return self._slices
+
+    @slices.setter
+    def slices(self, value):
+
+        if len(value) != self._original_data.ndim:
+            raise ValueError("The 'indices' tuple should have length {0}"
+                             .format(self._original_data.ndim))
+
+        # For now we require the indices to be in the same position, i.e. we
+        # don't allow changes in dimensionality of the derived dataset.
+        if hasattr(self, '_slices'):
+            has_changed = False
+            for idim in range(self._original_data.ndim):
+                before, after = self._slices[idim], value[idim]
+                if type(before) != type(after):
+                    raise TypeError("Can't change where the ``None`` values are in slices")
+                elif before != after:
+                    has_changed = True
+        else:
+            has_changed = False
+
+        self._slices = value
+
+        # Compute a subset state that represents the indexing - this is used
+        # for compute_statistic and compute_histogram
+        slice_labels = [slice(x) if x == 0 else x for x in self._slices]
+        self._slices_subset_state = SliceSubsetState(self._original_data, slice_labels)
+
+        # Construct a list of original pixel component IDs
+        self._original_pixel_cids = []
+        for idim in range(self._original_data.ndim):
+            if self._slices[idim] is None:
+                self._original_pixel_cids.append(self._original_data.pixel_component_ids[idim])
+
+        # Construct a list of original world component IDs
+        self._original_world_cids = []
+        if len(self._original_data.world_component_ids) > 0:
+            idim_new = 0
+            for idim in range(self._original_data.ndim):
+                if self.slices[idim] is None:
+                    self._cid_to_original_cid[self.world_component_ids[idim_new]] = \
+                        self._original_data.world_component_ids[idim]
+                    idim_new += 1
+
+        # Tell glue that the data has changed
+        if has_changed and self.hub is not None:
+            msg = NumericalDataChangedMessage(self)
+            self.hub.broadcast(msg)
+
+    @property
+    def label(self):
+        slice = '(' + ','.join([':' if x == 0 else str(x) for x in self._slices]) + ')'
+        return self._original_data.label + slice
+
+    @property
+    def shape(self):
```

Comment:
For now, this and several other properties below will just return the same as the original data, right?

Reply:
Yes, think so